### DISCUSSION TITLE
indexing gets the date of photo wrong

### DISCUSSION THREAD

**Discussion by agsha** [photoprism/photoprism#3114] [Q&A] [answered]
#### 1. What is not working as documented?
When I index my photos folder, around 10% of the photos date is indexed wrongly. They seem to all default to 08-dec-2002 although the photo was taken on another date. I have verified that the metadata contains the date taken. Other photo libraries like shotwell etc dont seem to have this issue

#### 2. How can we reproduce it?
Attaching the problematic file

1. Put the file in a folder
2. index it.
3. Observe that the date is 08-dec-2002 even though the photo was taken in 2013

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64 Build [220901-f493607b0]

(b) Database Type & Version: MySQL, SQLite

(c) Operating System Types & Versions: Ubuntu

(d) Browser Types & Versions: Chrome

(e) Ad Blockers, Browser Plugins, and/or Firewall Software? None

You can find the version/build number of the app in *Settings* by scrolling to the bottom. Note that MySQL 8 support has been discontinued, see system requirements at <https://docs.photoprism.app/getting-started/#system-requirements>.

*Always provide database and operating system details if it is a backend, import, or indexing issue. Should it be a frontend issue, at a minimum we require you to provide web browser and operating system details. When reporting a performance problem, we ask that you provide us with complete information about your environment, as there may be more than one cause.*

#### 7. On what kind of device is PhotoPrism installed?

regular desktop pc

(a) Device / Processor Type: AMD threadripper 3960x

(b) Physical Memory & Swap Space in GB 256GB RAM

(c) Storage Type: HDD, SSD, RAID, USB, Network Storage, 1TB nvme

(d) Anything else that might be helpful to know?

**Comment by lastzero** [comment]
Please upgrade to our latest release and test again. Thank you!

**Comment by agsha** [comment]
can confirm that same issue exists in Build [221118-e58fee0fb]

**Comment by graciousgrey** [comment]
December 8th 2002 is the CreateDate of your photos metadata. Do you have any idea why it differs from the DateTimeOriginal ? 

```
  "DateTimeOriginal": "2013:07:12 16:54:27",
  "CreateDate": "2002:12:08 12:00:00",

```

**Comment by lastzero** [comment]
Note that `CreateDate` is an XMP field and that Shotwell does not seem to have XMP support yet (which is not a bug in PhotoPrism):
- https://gitlab.gnome.org/GNOME/shotwell/-/issues/98

**Comment by agsha** [comment]
FWIW, I havent edited these photos. They were copied directly from the mobile phone to my file server. The issue seems to occur with many different phones over the years.

**Comment by graciousgrey** [comment, marked as answer]
This seems to be a bug in Android: https://issuetracker.google.com/issues/36967504

**Reply by graciousgrey** [reply]
You could use exiftool to correct the wrong date. A normal indexing should pick up the date change afterwards.

**Comment by agsha** [comment]
wow, nice job finding the issue.. Indeed all those photos were from different nexus cameras. I'll use the exiftool to correct the dates. Thanks so much. 
P.S. love photoprism.. fantastic app

**Comment by reidprichard** [comment]
Hmm, I'm having the same exact issue (photos showing 8 Dec 2002). However, I downloaded one and its metadata shows the correct date (12 May 2015), as does the associated JSON and the original file on Google Photos. I'm re-indexing my library to see if that might help.

Hmm, while indexing I'm noticing a TON of messages in the logs about an invalid local time offset: `time="2023-02-24T00:58:10Z" level=info msg="metadata: IMG_20150604_134047.jpg has an invalid local time offset (109465h40m50s)"`

**Reply by graciousgrey** [reply]
What metadata do you see when running `exiftool -j filename.jpg | grep Date` ?

**Reply by reidprichard** [reply]
Hmm, it appears that "GpsDateStamp", "DateTimeOriginal", and "GPSDateTime" all point to the correct value, while "CreateDate" shows 8 Dec 2002. Is it possible to index based on one of the other date values?
```
  "FileModifyDate": "2023:02:23 19:54:17-05:00",
  "FileAccessDate": "2023:02:24 09:13:57-05:00",
  "FileCreateDate": "2023:02:23 19:54:17-05:00",
  "GPSDateStamp": "2015:05:12",
  "CreateDate": "2002:12:08 12:00:00",
  "DateTimeOriginal": "2015:05:12 13:33:17",
  "GPSDateTime": "2015:05:12 17:33:13Z",
```

**Reply by lastzero** [reply]
Yes, we may be able to work around this. However it's not a bug in our software but in Android:

- https://forum.fairphone.com/t/bug-report-pictures-have-constant-exif-createdate-2002-12-08/25963
- https://issuetracker.google.com/issues/36967504?pli=1

**Reply by lastzero** [reply]
@reidprichard  @agsha You are invited to test these changes:

- #3229

**Reply by reidprichard** [reply]
Good to know! If worse comes to worst I could always download the affected images, batch modify the CreateDate, and replace the files on Photoprism with the modified versions. 
I'll update and post back as soon as the latest build is pushed out.

**Reply by agsha** [reply]
I have tested the developer preview with around 50 problematic photos. I am able to confirm that all the dates are now indexed properly! thanks so much!

**Comment by agsha** [comment]
If anyone else encounters this issue, just noting the exiftool command to fix all the dates.
This is because OS like truenas comes with a builtin deployment of photoprism

````
# don't forget the single quotes around CreateDate...
./exiftool -o  /path/to/folder/%f_modified.%e '-CreateDate<DateTimeOriginal' /path/to/problem_files_dir
````

explanation

`-o` -  force exiftool to write the modified file as a copy
`%f` - special character denoting the original filename
`%e` - special character denoting the extension

`'-CreateDate<DateTimeOriginal'` - tells exiftool to copy `DateTimeOriginal` into `CreateDate` (which is problematic)
